cmd/crates/soroban-spec-typescript/ts-tests/initialize.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+set +x
+
+# read .env file, but prefer explicitly set environment variables
+IFS=$'\n'
+for l in $(cat .env); do
+    IFS='=' read -ra VARVAL <<< "$l"
+    # If variable with such name already exists, preserves it's value
+    eval "export ${VARVAL[0]}=\${${VARVAL[0]}:-${VARVAL[1]}}"
+done
+unset IFS
+
+echo Network
+echo "  RPC:        $SOROBAN_RPC_URL"
+echo "  Passphrase: \"$SOROBAN_NETWORK_PASSPHRASE\""
+
+exe() { echo"${@/eval/}" ; "$@" ; }
+
+function main() {
+  curl -X POST "http://localhost:8000/soroban/rpc" || { echo "Make sure you're running standalone RPC network on localhost:8000" && exit 1; }
+  fund 
+  deploy
+  initialize
+  bindings
+}
+function fund_root() {
+  exe eval "./soroban config identity fund"
+}
+function fund_alice() {
+  exe eval "./soroban config identity generate alice"
+  exe eval "./soroban config identity fund alice"
+}
+function fund() {
+  fund_root
+  fund_alice
+}
+function bindings_custom_types() {
+  exe eval "./soroban contract bindings typescript --contract-id $(cat contract-id-custom-types.txt) --output-dir ./node_modules/test-custom-types --overwrite"
```

Comment:
why `exe eval`? Also seems like you could break bindings into it's own function and then below call with relevant args instead of a bunch of functions.

Reply:
Yeah but why do you need `exe` at all? why not `./soroban ...`

Reply:
To output the command that's going to be run. It ends up looking like this:

```
npm i

> postinstall
> ./initialize.sh

Network
  RPC:        http://localhost:8000/soroban/rpc
  Passphrase: "Standalone Network ; February 2017"
./soroban config identity fund
./soroban config identity generate alice
./soroban config identity fund alice
2023-10-23T15:10:42.831597Z  WARN soroban_cli::commands::config::network: Account already exists
(./soroban contract deploy --wasm ../../../../target/wasm32-unknown-unknown/test-wasms/test_custom_types.wasm) > contract-id-custom-types.txt
(./soroban contract deploy --wasm ../../../../target/wasm32-unknown-unknown/test-wasms/test_hello_world.wasm) > contract-id-hello-world.txt
(./soroban contract deploy --wasm ../../../../target/wasm32-unknown-unknown/test-wasms/test_swap.wasm) > contract-id-swap.txt
(./soroban contract install --wasm ../../../../target/wasm32-unknown-unknown/test-wasms/test_token.wasm) > contract-token-hash.txt
(./soroban contract deploy --wasm-hash 07ec9b8333159ac477239ff1a54c6fc45c5817e03cf0f45b6c9e51727c0e3dc7) > contract-id-token-a.txt
(./soroban contract deploy --wasm-hash 07ec9b8333159ac477239ff1a54c6fc45c5817e03cf0f45b6c9e51727c0e3dc7) > contract-id-token-b.txt
./soroban contract invoke --id CCE4XXFTF2JJRO3GK4NDEPIZGKZQYXEZUKXIVGDEULUEQZHDD4XOOTGK -- initialize --admin GDIY6AQQ75WMD4W46EYB7O6UYMHOCGQHLAQGQTKHDX4J2DYQCHVCR4W4 --decimal 0 --name 'Token A' --symbol 'A'

./soroban contract invoke --id CA63EAYAR75XGN62FLHTYRDKDQSCN2Q7XPNIZ635L3W5RV6GO3BKB4ET -- initialize --admin GDIY6AQQ75WMD4W46EYB7O6UYMHOCGQHLAQGQTKHDX4J2DYQCHVCR4W4 --decimal 0 --name 'Token B' --symbol 'B'

./soroban contract bindings typescript --contract-id CDSASQFKGWGL3HUWL6MRWSQMXC4PLCSYQRJYCDGNHKWQWRCQ7V7PLRW2 --output-dir ./node_modules/test-custom-types --overwrite

added 26 packages, and audited 27 packages in 4s

5 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> test-custom-types@0.0.0 build
> node ./scripts/build.mjs

./soroban contract bindings typescript --contract-id CDBUYGC6MLYU4LNB63THZQJOVVD65GUKWVGKAOOVVQ5E5MHOHEVXAPTB --output-dir ./node_modules/test-hello-world --overwrite

added 26 packages, and audited 27 packages in 1s

5 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> test-hello-world@0.0.0 build
> node ./scripts/build.mjs

./soroban contract bindings typescript --contract-id CCLE6DN54RHRC4P5TKPIMRUMC3LNCPYJBPL7D5RDEKAOB64JLQ25MWIO --output-dir ./node_modules/test-swap --overwrite

added 26 packages, and audited 27 packages in 1s

5 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> test-swap@0.0.0 build
> node ./scripts/build.mjs

./soroban contract bindings typescript --contract-id CCE4XXFTF2JJRO3GK4NDEPIZGKZQYXEZUKXIVGDEULUEQZHDD4XOOTGK --output-dir ./node_modules/token --overwrite

added 26 packages, and audited 27 packages in 900ms

5 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> token@0.0.0 build
> node ./scripts/build.mjs


removed 108 packages, and audited 191 packages in 48s

58 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

Reply:
Oh I see, you can also do `set -v` at the top of the file. Then using a functions it can be a bit more tidy:
```bash
function bind() {
  ./soroban contract bindings typescript --contract-id "$(cat $1)" --output-dir "./node_modules/$2" --overwrite
}


function bindings() {
  bind contract-id-custom-types.txt test-custom-types
  bind contract-id-hello-world.txt test-hello-world
  bind contract-id-swap.txt test-swap
  bind contract-id-token-a.txt token
}
```

Reply:
Oh nvm that also prints all the functions.

Reply:
I cleaned it up with some functions and added comments + links above the `exe` function definition. I think it looks much tidier; thanks! lmk if you have other ideas for cleaning it up.